### PR TITLE
[Keyboard] Add Mojo68 for Melgeek.

### PR DESCRIPTION
**Pull request by mj-bomb**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/melgeek/mojo68/rev1/rev1.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/melgeek/mojo68/rev1/rev1.c:164]: Not needed.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by mj-bomb** [comment]
@tzarc @fauxpark   Hello, my friend. Can you vet it for me

**Comment by mj-bomb** [comment]
@tzarc @fauxpark Hello, my friend. Could you mind review for my PR?

File: keyboards/melgeek/mojo68/rev1/rules.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
+RGB_MATRIX_DRIVER = IS31FL3741
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
```
Comment (fauxpark) [inline keyboards/melgeek/mojo68/rev1/rules.mk:23]:
```suggestion
```

File: keyboards/melgeek/mojo68/rev1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/melgeek/mojo68/rev1/rules.mk:4]:
```suggestion

# Bootloader selection
BOOTLOADER = atmel-dfu
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**